tests/unit/experimental/pipeline/test_pipeline.py

```diff
@@ -97,6 +97,25 @@ def test_pipeline_parameter_validation_one_component_all_good() -> None:
     assert is_valid is True
 
 
+def test_pipeline_parameter_validation_called_twice() -> None:
+    pipe = Pipeline()
+    component_a = ComponentPassThrough()
+    component_b = ComponentPassThrough()
+    pipe.add_component(component_a, "a")
+    pipe.add_component(component_b, "b")
+    pipe.connect("a", "b", {"value": "a.result"})
+    is_valid = pipe.validate_parameter_mapping_for_task(pipe.get_node_by_name("b"))
+    assert is_valid is True
+    with pytest.raises(PipelineDefinitionError):
+        pipe.validate_parameter_mapping_for_task(pipe.get_node_by_name("b"))
+    pipe.invalidate()
+    assert pipe.is_validated is False
+    assert len(pipe.param_mapping) == 0
+    assert len(pipe.missing_inputs) == 0
```

Comment:
Doesn't this only test that `pipe.invalidate()` gets called? In theory we don't need these assertions here right?

Reply:
Yes could remove them

Reply:
Maybe better: https://github.com/neo4j/neo4j-graphrag-python/pull/124/commits/f228dbf7e01cf0f91a318a2772b819e8fd0d84ac